core/src/test/java/io/github/tcdl/RequesterTest.java

```diff
@@ -152,7 +152,8 @@ private Requester initRequesterForResponsesWithTimeout(int numberOfResponses) th
         when(channelManagerMock.findOrCreateProducer(anyString())).thenReturn(producerMock);
 
         MsbContext msbContext = TestUtils.createSimpleMsbContext();
-        msbContext.setChannelManager(channelManagerMock);
+        msbContext = TestUtils.createMsbContext(msbContext.getMsbConfig(), msbContext.getMessageFactory(), 
```

Comment:
MsbContext couldn't provide setters accessible for the end-user.So, setters were removed from the class. Mock implementations of the aggregated objects can be added either through Constructor directly or with using Builder pattern. The simplest approach with Constructor has been implemented.